src/hotspot/share/runtime/threads.cpp

```diff
@@ -1081,7 +1081,7 @@ void Threads::remove(JavaThread* p, bool is_daemon) {
   ObjectSynchronizer::dec_in_use_list_ceiling();
 
   // Since Events::log uses a lock, we grab it outside the Threads_lock
-  Events::log(p, "Thread exited: " INTPTR_FORMAT, p2i(p));
+  Events::log(Thread::current(), "exited thread: " INTPTR_FORMAT, p2i(p));
```

Comment:
Most of the time a thread "exits" itself so `Thread::current() == p` in that case. Only in two error cases is a newly created thread removed by the thread that added it. So this revised wording doesn't really make sense for the majority of cases. I think it makes sense to report the thread doing the add/remove as well as the thread being added/removed, so I would suggest replacing `exited` with `removed`.

That said any change here may disrupt existing log parsing, including tests, so that needs to be considered.

Reply:
I adjusted the coding.